### PR TITLE
Google auth startup activity

### PR DESCRIPTION
**Pull request by Luca-Engel**
At start-up, ¨the user has three possibilities: sign in, sign out, or delete. All the buttons then call google authentication. 

If sign-in is called, then the method 'private fun launchPostLoginActivity(email : String)' in MainActivity.kt is called with the user's email passed as an argument.

Test coverage is not that high because I was not able to test all the google authentication parts (e.g., sign in)

MockGoogleAuthenticator theoretically not necessary but once we add a check in MainActivity to see if user already is logged in and then skip the manual sign in part, it will be useful for testing

### CODE REVIEW

File: app/src/main/java/com/github/sdpcoachme/firebase/database/RealDatabase.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.github.sdpcoachme.firebase.database
+
+class RealDatabase : Database {
+
+    override fun isExistingUser(email: String): Boolean {
+        return true
+    }
```
Comment (Ayskin) [inline app/src/main/java/com/github/sdpcoachme/firebase/database/RealDatabase.kt:7]: Maybe we can add a TODO here if we need to implement this function later ?

File: app/src/androidTest/java/com/github/sdpcoachme/MainActivityTest.kt [hunk 2]
```diff
@@ -1,34 +1,136 @@
...
+//        Intents.init()
+//
+//        val name = "John Lennon"
+//        composeTestRule.onNodeWithTag("textfield").performTextInput(name)
+//        Espresso.closeSoftKeyboard()
+//        composeTestRule.onNodeWithTag("button").performClick()
+//
+//        intended(allOf(hasComponent(GreetingActivity::class.java.name), hasExtra("name", name)))
+//        Intents.release()
+//    }
```
Comment (Ayskin) [inline app/src/androidTest/java/com/github/sdpcoachme/MainActivityTest.kt:33]: I think we can delete the commented correctIntentFiredOnDisplayMessageClick() test function !

**Review by Ayskin — COMMENTED** [review]
Seems all good and clean to me !

File: app/src/main/java/com/github/sdpcoachme/firebase/database/RealDatabase.kt (hunk 1 above)
Comment (lindsaybordier) [reply app/src/main/java/com/github/sdpcoachme/firebase/database/RealDatabase.kt:7]: I agree with that

File: app/src/main/java/com/github/sdpcoachme/MainActivity.kt [hunk 3]
```diff
@@ -4,53 +4,112 @@ import android.content.Intent
...
+            res,
+            { email -> launchPostLoginActivity(email ?: "") },
+            { errorMsg -> signInInfo = errorMsg.toString() }
+        )
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        authenticator = (application as CoachMeApplication).authenticator
+        this.database =  (application as CoachMeApplication).database
```
Comment (lindsaybordier) [inline app/src/main/java/com/github/sdpcoachme/MainActivity.kt:44]: This is great a very clean use of dependency injection and will merge easily in the project as it was coded the same way in the branch implementing the database.

File: app/src/androidTest/java/com/github/sdpcoachme/firebase/auth/GoogleAuthenticatorTest.kt [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Test
+    fun isSignedInReturnsFalseIfUserNotSignedIn() {
+        FirebaseAuth.getInstance().signOut()
+        assertThat(GoogleAuthenticator().isSignedIn(), `is`(false))
+    }
+
+    @Test
+    fun signIt() {
```
Comment (lindsaybordier) [inline app/src/androidTest/java/com/github/sdpcoachme/firebase/auth/GoogleAuthenticatorTest.kt]: This test seems empty so we could maybe remove it

**Pull request merged**